astro/sessions.go

```diff
@@ -169,6 +186,13 @@ func (s *Session) applyWithGraph(boundExecutions []*boundExecution) (<-chan stri
 	status := make(chan string, numberOfExecutions*10)
 	results := make(chan *Result, numberOfExecutions)
 
+	isCancelled := false
```

Comment:
Discussed this offline and we're going to move this out of the astro package completely and do it at a process level.